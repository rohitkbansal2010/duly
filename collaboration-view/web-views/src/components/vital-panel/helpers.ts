import { measurementsUnitsMap } from '@constants';
import { MeasurementType } from '@enums';

const isToday = (date: Date): boolean => {
  const today = new Date();

  const year = today.getFullYear();
  const month = today.getMonth();
  const day = today.getDate();

  return date.getFullYear() === year &&
    date.getMonth() === month &&
    date.getDate() === day;
};

export const getUnitMeasurement = (measurementType: MeasurementType, unit?: string): string => {
  if (measurementType === MeasurementType.SYSTOLIC_BLOOD_PRESSURE) {
    return measurementsUnitsMap[MeasurementType.DIASTOLIC_BLOOD_PRESSURE][unit as string];
  }

  return measurementType !== MeasurementType.PAIN_LEVEL
    ? measurementsUnitsMap[measurementType][unit as string]
    : '' ;
};

export const getTextMeasurement = (measurementType: MeasurementType, date: string): string => {
  switch (measurementType) {
    case MeasurementType.BODY_HEIGHT:
      return 'HEIGHT';
    case MeasurementType.BODY_WEIGHT:
      return 'WEIGHT';
    case MeasurementType.BODY_MASS_INDEX:
      return 'AGE';

    default:
      return isToday(new Date(date)) ? `TODAY's` : 'MOST RECENT';
  }
};
