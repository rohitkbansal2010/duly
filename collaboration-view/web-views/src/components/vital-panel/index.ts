export { VitalPanel } from './vital-panel';
