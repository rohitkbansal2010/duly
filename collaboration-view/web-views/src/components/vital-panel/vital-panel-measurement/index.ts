export { VitalPanelMeasurement } from './vital-panel-measurement';
export type { VitalPanelMeasurementPropsType } from './vital-panel-measurement';
