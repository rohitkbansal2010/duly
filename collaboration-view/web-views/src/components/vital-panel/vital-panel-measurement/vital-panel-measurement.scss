@import "~@styles/main.scss";
@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.vitalPanelMeasurementContainer {
  display: flex;
  align-items: center;
  padding: 1.25rem 0 1.25rem 1rem;

  &:not(:first-child) {
    border-left: 0.06rem solid $iron;
  }

  &Icon {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 3.75rem;
    height: 3.75rem;
    margin-right: 1rem;
    border-radius: 50%;
    @include background-opacity($dupage-magenta, 0.3);

    & img {
      object-fit: cover;
    }

    &BloodPressure {
      width: 2.4rem;
      height: 2.02rem;
    }

    &HeartRate {
      width: 2.29rem;
      height: 2.1rem;
    }

    &Temperature {
      width: 1.38rem;
      height: 2.5rem;
    }

    &RespiratoryRate {
      width: 2.29rem;
      height: 2.29rem;
    }

    &PainLevel {
      width: 1.16rem;
      height: 2.06rem;
    }
  }

  &AdditionalInfo {
    &TopBlock,
    &BottomBlock {
      display: flex;
    }

    &TopBlock {
      align-items: center;
    }

    &BottomBlock {
      align-items: baseline;
    }

    &Text {
      color: $medium-gray;
      font-size: var(--font-size-16px-4px);
      line-height: 1.5rem;
      @include nexa-heavy-highlight;
    }

    &Date {
      margin-left: 0.75rem;
      color: $dupage-violet;
      font-size: var(--font-size-14px-4px);
      line-height: 1rem;
      @include nexa-extra-bold-highlight;
    }

    &Value {
      min-height: 2.625rem;
      color: $dupage-magenta;
      font-size: var(--font-size-32px-4px);
      line-height: 2.625rem;
      @include nexa-heavy-highlight;
    }

    &Unit,
    &MaxScaleValue,
    &AdditionalValue {
      margin-left: 0.25rem;
      font-size: var(--font-size-22px-4px);
      line-height: 1.625rem;
      @include nexa-heavy-highlight;
    }

    &Unit,
    &MaxScaleValue {
      color: $dupage-magenta;
    }

    &AdditionalValue {
      color: $dupage-violet;
    }
  }
}
