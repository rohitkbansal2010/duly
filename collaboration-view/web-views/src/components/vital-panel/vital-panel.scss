@import "~@styles/colors.scss";

.vitalPanelContainer {
  display: flex;
  border: 0.06rem solid $iron;
  border-radius: 0.75rem;
}
