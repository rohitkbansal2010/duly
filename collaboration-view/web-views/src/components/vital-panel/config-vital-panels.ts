import { CSSProperties } from 'react';

import { vitalsChartsMap } from '@constants';
import { CardType, SortDirection } from '@enums';
import { ChartPoint } from '@interfaces';
import { DateTimeDoubleChartDataset, Measurement } from '@types';
import { calculateYears } from '@utils';

import { getTextMeasurement, getUnitMeasurement } from './helpers';

type GetConfigVitalPanelByTypeParameters = {
  datasets: DateTimeDoubleChartDataset[];
  cardType: CardType,
  measurements: Measurement[],
  birthDate?: string
};

type GetMeasurementsComponentsByTypeParameters = GetConfigVitalPanelByTypeParameters;

type GeneralAdviceComponentType = {
  title: string;
  text: string;
  style: CSSProperties;
};

export type MeasurementComponentType = {
  style?: CSSProperties;
  text?: string;
  value?: number ;
  date?: string;
  unit?: string;
  icon?: string;
  additionalValue?: number | string;
  maxScaleValue?: number;
  convertedValue?: string | number | undefined;
  convertedUnit?: string | undefined;
};

type ConfigVitalPanelType = {
  measurementsComponents: MeasurementComponentType[];
  generalAdviceComponent?: GeneralAdviceComponentType;
}

const generalAdvices = new Map([
  [ CardType.BLOOD_PRESSURE, {
    title: 'General Advice',
    text: 'Blood pressure numbers of less than 120/80 mm Hg are considered within the normal range. If your results fall into this category, stick with heart-healthy habits.',
    style: { width: '56%' },
  } ],
  [ CardType.HEART_RATE, {
    title: 'General Advice',
    text: 'A normal rate is usually started as 60 to 100 beats per minute. Slower than 60 is bradycardia ("slow heart"), faster than 100 is tachycardia ("fast heart"). But an ideal resting heart is closer to 50 to 70.',
    style: { width: '59%' },
  } ],
  [ CardType.TEMPERATURE, {
    title: 'General Advice',
    text: '97.5-98.8 F is generally normal reported range for normal body.',
    style: { width: '56%' },
  } ],
  [ CardType.RESPIRATORY_RATE, {
    title: 'General Advice',
    text: 'The normal respiratory rate for healthy adults is between 12 and 20 breaths per minute. Breathing rates of below 12 or above 20 can mean a disruption in normal breathing processes.',
    style: { width: '56%' },
  } ],
  [ CardType.PAIN_LEVEL, {
    title: 'General Advice',
    text: 'Report to your doctor if you\'re experiencing any aching, biting, burning, cramping, dull, gnawing, heavy, hot, piercing, pinching, sharp, shooting, sickening, sore, splitting, stabbing, tender, tingling or throbbing feeling.',
    style: { width: '61%' },
  } ],
  [ CardType.WEIGHT_HEIGHT, {
    title: 'General Advice',
    text: 'Daily weight fluctuation is normal.  The average adult\'s weight fluctuates up to 5 or 6 pounds per day. It all comes down to what and when you eat, drink, exercise, and even sleep.',
    style: { width: '48%' },
  } ],
]);

const getMeasurementsComponentsByType = ({
  datasets,
  measurements,
  cardType,
  birthDate,
}: GetMeasurementsComponentsByTypeParameters): MeasurementComponentType[] => {
  const [ measurementOne, measurementTwo, measurementThree ] = measurements;
  const {
    measurementType,
    measured,
    unit,
    maxScaleValue,
    value,
  } = measurementOne;

  const defaultMeasurementProps = {
    icon: vitalsChartsMap[cardType].icon,
    text: getTextMeasurement(measurementType, measured),
    date: measured,
    unit: getUnitMeasurement(measurementType, unit),
    additionalValue: cardType.includes(CardType.BLOOD_PRESSURE) ? measurements[1].value : '',
    value,
  };

  const xyValues = Array.from(
    datasets.map(({ data: { values } }) =>
      values)
      .flat()
      .map(({ x, y }) =>
        (
          {
            x,
            y,
          }
        ))
  );

  const sortDataByValuesToDirection = (
    data: ChartPoint[],
    direction = SortDirection.TO_MIN
  ): ChartPoint[] =>
    [ ...data ].sort((a, b) => {
      if (a.y === b.y) {
        return new Date(a.x).getMilliseconds() - new Date(b.x).getMilliseconds();
      }

      return direction === SortDirection.TO_MAX ? a.y - b.y : b.y - a.y;
    });

  const sortedDataByValuesToMax = sortDataByValuesToDirection(xyValues, SortDirection.TO_MAX);
  const sortedDataByValuesToMin = sortDataByValuesToDirection(xyValues);
    
  switch (cardType) {
    case CardType.BLOOD_PRESSURE:
      return [ {
        ...defaultMeasurementProps,
        style: { width: '44%' },
      } ];

    case CardType.BLOOD_OXYGEN:
      return [
        {
          ...defaultMeasurementProps,
          style: { width: '36%', padding: '1.25rem 0 1.25rem 3.125rem' },
        },
        {
          text: 'LOWEST',
          value: sortedDataByValuesToMax[0].y,
          style: { width: '33%', padding: '1.25rem 4rem' },
          date: sortedDataByValuesToMax[0].x,
          unit: '%',
        },
        {
          text: 'HIGHEST',
          value: sortedDataByValuesToMin[0].y,
          style: { width: '31%', padding: '1.25rem 3.125rem 1.25rem 4rem' },
          date: sortedDataByValuesToMin[0].x,
          unit: '%',
        },
      ];

    case CardType.HEART_RATE:
      return [ {
        ...defaultMeasurementProps,
        style: { width: '41%' },
      } ];

    case CardType.TEMPERATURE:
      return [ {
        ...defaultMeasurementProps,
        style: { width: '44%' },
      } ];

    case CardType.RESPIRATORY_RATE:
      return [ {
        ...defaultMeasurementProps,
        style: { width: '44%' },
      } ];

    case CardType.PAIN_LEVEL:
      return [ {
        ...defaultMeasurementProps,
        style: { width: '39%' },
        maxScaleValue,
      } ];

    case CardType.WEIGHT_HEIGHT:
      return [
        {
          ...defaultMeasurementProps,
          date: '',
          style: { minWidth: '26%', padding: '1.25rem 5.56rem 1.25rem 2.75rem' },
        },
        {
          text: measurementTwo &&
            getTextMeasurement(measurementTwo.measurementType, measurementTwo.measured),
          date: '',
          value: measurementTwo && measurementTwo.value,
          unit: measurementTwo &&
            getUnitMeasurement(measurementTwo.measurementType, measurementTwo.unit),
          style: { minWidth: '26%', padding: '1.25rem 2.75rem' },
        },
      ];

    case CardType.BODY_MASS_INDEX:
      return [
        {
          ...defaultMeasurementProps,
          date: '',
          value: calculateYears(birthDate, measured),
          unit: 'years',
          style: { width: '38%', padding: '1.25rem 4rem 1.25rem 7.25rem' },
        },
        {
          text: measurementTwo &&
            getTextMeasurement(measurementTwo.measurementType, measurementTwo.measured),
          date: '',
          value: measurementTwo && measurementTwo.value,
          unit: measurementTwo &&
            getUnitMeasurement(measurementTwo.measurementType, measurementTwo.unit),
          style: { minWidth: '32%', padding: '1.25rem 4.2rem 1.25rem 4rem' },
        },
        {
          text: measurementThree &&
            getTextMeasurement(measurementThree.measurementType, measurementTwo.measured),
          date: '',
          value: measurementThree && measurementThree.value,
          unit: measurementThree &&
            getUnitMeasurement(measurementThree.measurementType, measurementThree.unit),
          style: { minWidth: '30%', padding: '1.25rem 7.25rem 1.25rem 4rem' },
        },
      ];
  }
};

export const getConfigVitalPanelByType = ({
  datasets,
  cardType,
  measurements,
  birthDate = '',
}: GetConfigVitalPanelByTypeParameters): ConfigVitalPanelType =>
  ({
    measurementsComponents:
      getMeasurementsComponentsByType({
        datasets,
        cardType,
        measurements,
        birthDate,
      }),
    generalAdviceComponent: generalAdvices.get(cardType),
  });
