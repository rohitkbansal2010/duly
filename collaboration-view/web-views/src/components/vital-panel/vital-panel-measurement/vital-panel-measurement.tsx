import { CardType } from '@enums';
import { formatMDYYYYDate } from '@utils';

import { MeasurementComponentType } from '../config-vital-panels';

import styles from './vital-panel-measurement.scss';

export type VitalPanelMeasurementPropsType = MeasurementComponentType & {
  cardType: CardType;
  isVisible: boolean;
  convertedHeightUnit: string | null;
  convertedWeightUnit: string | null
  convertedHeightValue?: string | null;
  convertedWeightValue?: number | null;
  toggle: string;
};

export const VitalPanelMeasurement = ({
  cardType,
  isVisible,
  text,
  value,
  convertedWeightValue,
  convertedHeightValue,
  convertedHeightUnit,
  convertedWeightUnit,
  style,
  unit,
  maxScaleValue,
  additionalValue,
  icon,
  date,
}: VitalPanelMeasurementPropsType) => {
  if (!isVisible) return null;
  
  return (
    
    
    <div className={styles.vitalPanelMeasurementContainer} style={style}>     
      {icon && (
        <div className={styles.vitalPanelMeasurementContainerIcon}>
          <img
            src={icon}
            alt="vital icon"
            className={styles[`vitalPanelMeasurementContainerIcon${cardType}`]}
            date-test="vital-panel-measurement-icon"
          />
        </div>
      )}
      <div className={styles.vitalPanelMeasurementContainerAdditionalInfo}>
        <div className={styles.vitalPanelMeasurementContainerAdditionalInfoTopBlock}>
          <div
            className={styles.vitalPanelMeasurementContainerAdditionalInfoText}
            date-test="vital-panel-measurement-text"
          >
            {text}
          </div>
          {date && (
            <div
              className={styles.vitalPanelMeasurementContainerAdditionalInfoDate}
              date-test="vital-panel-measurement-date"
            >
              {formatMDYYYYDate(new Date(date))}
            </div>
          )}
        </div>
        <div className={styles.vitalPanelMeasurementContainerAdditionalInfoBottomBlock}>
          <div
            className={styles.vitalPanelMeasurementContainerAdditionalInfoValue}
            date-test="vital-panel-measurement-value"
          >
            {text === 'WEIGHT' && convertedWeightValue ? convertedWeightValue : 
            text === 'HEIGHT' && convertedHeightValue ? convertedHeightValue : value}
          </div>
          {!additionalValue && unit && (
            <div
              className={styles.vitalPanelMeasurementContainerAdditionalInfoUnit}
              date-test="vital-panel-measurement-unit"
            >
              {text === 'WEIGHT' && convertedWeightUnit ? convertedWeightUnit : 
            text === 'HEIGHT' && convertedHeightUnit ? '' : unit}
            </div>
          )}
          {maxScaleValue && (
            <div
              className={styles.vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue}
              date-test="vital-panel-measurement-max-scale-value"
            >
              {`/ ${maxScaleValue}`}
            </div>
          )}
          {additionalValue && (
            <>
              <div
                className={styles.vitalPanelMeasurementContainerAdditionalInfoAdditionalValue}
                date-test="vital-panel-measurement-additional-value"
              >
                {`/ ${additionalValue}`}
              </div>
              <div className={styles.vitalPanelMeasurementContainerAdditionalInfoUnit}>
                {unit}
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
};
