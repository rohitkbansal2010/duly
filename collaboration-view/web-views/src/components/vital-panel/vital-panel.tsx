import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';

import { CardType } from '@enums';
import { RootState } from '@redux/reducers';
import { DateTimeDoubleChartDataset, Vital } from '@types';
import { convertHeightToFeet, convertWeightToLBS } from '@utils';

import { getConfigVitalPanelByType } from './config-vital-panels';
import { VitalPanelGeneralAdvice } from './vital-panel-general-advice';
import { VitalPanelMeasurement } from './vital-panel-measurement';
import styles from './vital-panel.scss';

export type VitalPanelPropsType = {
  toggle: string;
  cardType: CardType;
  vitals: Vital[];
};

export const VitalPanel = ({
  toggle,
  cardType,
  vitals,
}: VitalPanelPropsType) => {
  const patientData = useSelector(({ CURRENT_APPOINTMENT }: RootState) =>
    CURRENT_APPOINTMENT.patientData);
  const { datasets } = useSelector(({ OVERVIEW_WIDGETS }: RootState) =>
    OVERVIEW_WIDGETS.chart) || { datasets: [] as DateTimeDoubleChartDataset[] };

  const measurements = vitals.map(({ measurements }) =>
    measurements).flat();
  const {
    measurementsComponents,
    generalAdviceComponent,
  } = getConfigVitalPanelByType({
    datasets,
    cardType,
    measurements,
    birthDate: patientData?.birthDate,
  });
  const [ convertedHeightValue, setConvertedHeightValue ] = useState<string | null>(null);
  const [ convertedWeightValue, setConvertedWeightValue ] = useState<number | null>(null);
  const [ convertedWeightUnit, setConvertedWeightUnit ] = useState<string | null>(null);
  const [ convertedHeightUnit, setConvertedHeightUnit ] = useState<string | null>(null);
  
  useEffect(() => {
    if(toggle === 'off' && (cardType === 'WeightAndHeight' || cardType === 'BodyMassIndex')) {
      const heightAtIndex = vitals.findIndex(ele => 
        ele.vitalType === 'Height');
      const weightAtIndex = vitals.findIndex(ele => 
        ele.vitalType === 'Weight');
      setConvertedHeightValue(convertHeightToFeet(vitals[heightAtIndex].measurements[0].value));
      setConvertedWeightValue(convertWeightToLBS(vitals[weightAtIndex].measurements[0].value));
      setConvertedWeightUnit('lbs');
      setConvertedHeightUnit('inch');
    }
    else if(toggle === 'on' && (cardType === 'WeightAndHeight' || cardType === 'BodyMassIndex')){
      setConvertedHeightValue(null);
      setConvertedWeightValue(null);
      setConvertedWeightUnit(null);
      setConvertedHeightUnit(null);
    }
  }, [ toggle, cardType ]);

  return (
    <div className={styles.vitalPanelContainer}>
      {measurementsComponents.map((measurementComponent, index) =>
        (
          <>
            <VitalPanelMeasurement
              key={index}
              cardType={cardType}
              isVisible={!!measurementComponent.value}
              toggle={toggle}
              convertedHeightValue={convertedHeightValue}
              convertedWeightValue={convertedWeightValue}
              convertedWeightUnit={convertedWeightUnit}
              convertedHeightUnit={convertedHeightUnit}
              {...measurementComponent}
            /></>
        ))}
      {generalAdviceComponent && (
        <VitalPanelGeneralAdvice {...generalAdviceComponent} />
      )}
    </div>
  );
};
