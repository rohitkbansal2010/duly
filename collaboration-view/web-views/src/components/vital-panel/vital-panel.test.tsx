import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as redux from 'react-redux';

import { CardType, MeasurementType, VitalType } from '@enums';
import { DateTimeDoubleChartDataset } from '@types';

import { getConfigVitalPanelByType } from './config-vital-panels';
import { VitalPanel, VitalPanelPropsType } from './vital-panel';
import { VitalPanelMeasurementPropsType } from './vital-panel-measurement';

const mockProps: VitalPanelPropsType = {
  toggle: 'off',
  cardType: CardType.BLOOD_PRESSURE,
  vitals: [
    {
      vitalType: VitalType.BLOOD_PRESSURE,
      measurements: [
        {
          measurementType: MeasurementType.SYSTOLIC_BLOOD_PRESSURE,
          value: 135.14,
          measured: '2021-12-01T11:52:37.583195+00:00',
          unit: 'mm[Hg]',
        },
        {
          measurementType: MeasurementType.DIASTOLIC_BLOOD_PRESSURE,
          value: 93.07,
          measured: '2021-12-01T11:52:37.583195+00:00',
          unit: 'mm[Hg]',
        },
      ],
    },
  ],
};
const mockPatientData = { birthDate: '1979-09-09' };
const mockChart = { datasets: [ { data: { values: [ { y: 94 }, { y: 95 }, { y: 98 } ] } } ] };

const setUp = (props: VitalPanelPropsType) =>
  shallow(<VitalPanel {...props} />);

describe('VitalPanel', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((selector: (state: unknown)=>unknown) =>
        selector({
          CURRENT_APPOINTMENT: { patientData: mockPatientData },
          OVERVIEW_WIDGETS: { chart: mockChart },
        }));

    component = setUp(mockProps);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render the VitalPanelMeasurement components', () => {
    const mockMeasurements = mockProps.vitals.map(({ measurements }) =>
      measurements).flat();
    const { measurementsComponents } =
      getConfigVitalPanelByType({
        datasets: mockChart.datasets as DateTimeDoubleChartDataset[],
        cardType: mockProps.cardType,
        measurements: mockMeasurements,
        birthDate: mockPatientData.birthDate,
      });

    const vitalPanelMeasurementComponents = component.find('VitalPanelMeasurement');

    expect(vitalPanelMeasurementComponents).toHaveLength(measurementsComponents.length);
    vitalPanelMeasurementComponents.forEach((vitalPanelMeasurementComponent, index) => {
      const expectedProps: VitalPanelMeasurementPropsType = {
        toggle: 'off',
        convertedHeightUnit: null,
        convertedWeightUnit: null,
        convertedHeightValue: null,
        convertedWeightValue: null,
        cardType: mockProps.cardType,
        isVisible: !!measurementsComponents[index].value,
        ...measurementsComponents[index],
      };

      expect(vitalPanelMeasurementComponent.props()).toEqual(expectedProps);
    });
  });

  it('should render the VitalPanelGeneralAdvice component, if generalAdviceComponent exists', () => {
    const mockMeasurements = mockProps.vitals.map(({ measurements }) =>
      measurements).flat();
    const { generalAdviceComponent } =
      getConfigVitalPanelByType({
        datasets: mockChart.datasets as DateTimeDoubleChartDataset[],
        cardType: mockProps.cardType,
        measurements: mockMeasurements,
        birthDate: mockPatientData.birthDate,
      });

    const vitalGeneralAdviceComponent = component.find('VitalPanelGeneralAdvice');

    expect(vitalGeneralAdviceComponent).toHaveLength(1);
    expect(vitalGeneralAdviceComponent.props()).toEqual(generalAdviceComponent);
  });

  it('should not render the VitalPanelGeneralAdvice component, if generalAdviceComponent does not exist', () => {
    const component = setUp({
      toggle: 'off',
      cardType: CardType.BLOOD_OXYGEN,
      vitals: [
        {
          vitalType: VitalType.BLOOD_OXYGEN,
          measurements: [
            {
              measurementType: MeasurementType.OXYGEN_SATURATION,
              value: 92.31,
              measured: '2021-12-01T11:52:37.5832382+00:00',
              unit: '%',
            },
          ],
        },
      ],
    });

    expect(toJson(component)).toMatchSnapshot();

    const vitalGeneralAdviceComponent = component.find('VitalPanelGeneralAdvice');
    expect(vitalGeneralAdviceComponent).toHaveLength(0);
  });
});
