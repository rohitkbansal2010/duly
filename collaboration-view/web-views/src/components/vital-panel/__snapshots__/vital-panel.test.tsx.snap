// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VitalPanel should not render the VitalPanelGeneralAdvice component, if generalAdviceComponent does not exist 1`] = `
<div
  className="vitalPanelContainer"
>
  <VitalPanelMeasurement
    additionalValue=""
    cardType="BloodOxygen"
    convertedHeightUnit={null}
    convertedHeightValue={null}
    convertedWeightUnit={null}
    convertedWeightValue={null}
    date="2021-12-01T11:52:37.5832382+00:00"
    isVisible={true}
    key="0"
    style={
      Object {
        "padding": "1.25rem 0 1.25rem 3.125rem",
        "width": "36%",
      }
    }
    text="MOST RECENT"
    toggle="off"
    unit="%"
    value={92.31}
  />
  <VitalPanelMeasurement
    cardType="BloodOxygen"
    convertedHeightUnit={null}
    convertedHeightValue={null}
    convertedWeightUnit={null}
    convertedWeightValue={null}
    isVisible={true}
    key="1"
    style={
      Object {
        "padding": "1.25rem 4rem",
        "width": "33%",
      }
    }
    text="LOWEST"
    toggle="off"
    unit="%"
    value={94}
  />
  <VitalPanelMeasurement
    cardType="BloodOxygen"
    convertedHeightUnit={null}
    convertedHeightValue={null}
    convertedWeightUnit={null}
    convertedWeightValue={null}
    isVisible={true}
    key="2"
    style={
      Object {
        "padding": "1.25rem 3.125rem 1.25rem 4rem",
        "width": "31%",
      }
    }
    text="HIGHEST"
    toggle="off"
    unit="%"
    value={98}
  />
</div>
`;

exports[`VitalPanel should render correctly 1`] = `
<div
  className="vitalPanelContainer"
>
  <VitalPanelMeasurement
    additionalValue={93.07}
    cardType="BloodPressure"
    convertedHeightUnit={null}
    convertedHeightValue={null}
    convertedWeightUnit={null}
    convertedWeightValue={null}
    date="2021-12-01T11:52:37.583195+00:00"
    icon="condition-magenta.svg"
    isVisible={true}
    key="0"
    style={
      Object {
        "width": "44%",
      }
    }
    text="MOST RECENT"
    toggle="off"
    unit="mmHg"
    value={135.14}
  />
  <VitalPanelGeneralAdvice
    style={
      Object {
        "width": "56%",
      }
    }
    text="Blood pressure numbers of less than 120/80 mm Hg are considered within the normal range. If your results fall into this category, stick with heart-healthy habits."
    title="General Advice"
  />
</div>
`;
